contributing/code/bc.rst

```diff
@@ -267,6 +275,11 @@ Turn static into non static                         No
 .. [4] When changing the parent class, the original parent class must remain an
        ancestor of the class.
 
+.. [5] The value of a constant may only be changed when the constants aren't
+       used in configuration (e.g. Yaml and XML files), as these do not support
+       constants and have to hardcode the value. For instance, event name
+       constants can't change the value without introducing a BC break.
```

Comment:
I think Nicolas' point is probably that if a value will likely be used in `serialize`/`unserialize`, like if there were some constant that was commonly put onto `User` classes (since those are serialized). But this does highlight that this is slippery. Perhaps that this should be re-worded that changing constant values is discouraged, but is allowed if we're quite sure that it's not being used in configuration files or in objects that are serialized. Anyways, the "serialized" is a good point, because it might make us think about this in the future and realize that changing some constant value would be bad in one case :)